src/conduit/ConduitClient.php

```diff
@@ -163,6 +184,55 @@ public function callMethod($method, array $params) {
     return $conduit_future;
   }
 
+  // UBER CODE BEGIN
+  private function addWonkaClaimToCookies($cookies) {
+    $claim = $this->getWonkaClaim();
+    $new_cookies = '';
+    if (empty($claim)) {
+      return $cookies;
+    }
+    $usso_cookie = "usso={$claim}";
+    if (empty($cookies)) {
+      $new_cookies = $usso_cookie;
+    } else {
+      $new_cookies = $cookies."; {$usso_cookie}";
+    }
+    return $new_cookies;
+  }
+
+  private function getWonkaClaim() {
+    $claim = '';
+    // if the file exists and is current, use it.
+    $claim_file = getenv('ARC_USSO_COOKIE');
+    if (file_exists($claim_file)) {
+      $stale = time() - filemtime($claim_file);
+      if ($stale < 60) {
+        $claim = $this->readClaimFile($claim_file);
+      }
+    }
+    if (empty($claim)) {
+      $wonka_cli = getenv('ARC_WONKA_CLI');
+      system("{$wonka_cli} {$claim_file}");
```

Comment:
There are plenty of ways to trash a user if you're allowed to write to that user's environment. This is not a security flaw, users control their env variables. If this was a setuid program, then yes this would be an issue. You can make exactly the same argument about using an executable in your prompt variable, or setting using the CC env variable in a makefile.